test/extended/openstack/loadbalancers.go

```diff
@@ -419,3 +437,59 @@ func createTestDeployment(depName string, labels map[string]string, replicas int
 	}}
 	return testDeployment
 }
+
+// Active wait (up to 3 minutes) until a certain number of members match a regex on status attribute.
+func waitUntilNmembersReady(client *gophercloud.ServiceClient, pool *pools.Pool, n int32, status string) error {
+
+	var resultingMembers []string
+	var err error
+
+	e2e.Logf("Waiting for %d members online in the pool %s", n, pool.Name)
+	err = wait.Poll(15*time.Second, 3*time.Minute, func() (bool, error) {
+		resultingMembers, err = getMembersOnStatus(client, pool, status)
+		if err != nil {
+			return false, err
+		}
+		if len(resultingMembers) == int(n) {
+			e2e.Logf("Found expected number of %s members, checking again 5 seconds later to confirm that it is stable", status)
+			time.Sleep(5 * time.Second)
+			resultingMembers, err = getMembersOnStatus(client, pool, status)
+			if err != nil {
+				return false, err
+			} else if len(resultingMembers) == int(n) {
+				e2e.Logf("Wait completed, %s members found: %q", status, resultingMembers)
+				return true, nil
+			}
+		}
+		e2e.Logf("Waiting for next iteration, %s members found: %q", status, resultingMembers)
+		return false, nil
+	})
+
+	if err != nil && strings.Contains(err.Error(), "timed out waiting for the condition") {
+		allMembers, err := pools.ListMembers(client, pool.ID, pools.ListMembersOpts{}).AllPages()
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("unexpected number of %s after active wait. All members status: %q", status, allMembers)
```

Comment:
If you want to print the existing members, the `%+v` formatting is probably more readable. I also suggest you print the number of expected members, and number or actual members.
```suggestion
		return fmt.Errorf("expected %d members matching %s status, got %d. All members: %+v", n, status, len(allMembers), allMembers)
```

Reply:
Done, thanks!